### PR TITLE
feat(tokens): add more fields to metadata

### PR DESCRIPTION
**Pull request by satish-ravi**
Fixes ACT-923, adds fields needed for the new token details screen. 

- `showZeroBalance` - true for tokens to be displayed when balance is 0
- `isStableCoin` - true for core stable tokens
- `isCashInEligible` - true for tokens that can be cashed-in
- `isCashOutEligible` - true for tokens that can be cashed-out
- `infoUrl` - link to coingecko
- `networkIconUrl` - computed, set for all non native tokens with the native token's image url

### CODE REVIEW

**Comment by cajubelt** [comment]
for network name could we just use a static mapping from network id to network name in app code?

**Comment by cajubelt** [comment]
hidePriceDelta is a little weird, should it be isStableCoin instead?

**Comment by MuckT** [comment]
@cajubelt I don't think we should do a static mapping as it doesn't account for chain forks e.g BTC > BTH.

**Comment by cajubelt** [comment]
@MuckT so if a chain forked what would you do? Seems like you would need app code changes no matter what to support both forks right?

**Comment by MuckT** [comment]
@cajubelt I am not sure we would need app code changes to support both forks with dynamic configs. Also if we did have a static mapping wouldn't it display the incorrect version on older app versions?

**Comment by cajubelt** [comment]
@MuckT to my knowledge, in past forks, one branch (I'm going to call this "the trunk", as opposed to "the fork") has kept the same name. Like Ethereum vs Ethereum Classic. Seems like it would take full-stack code changes to turn on support for the fork. Is it such a bad thing to just default to the trunk and add support for the fork in a later version? Seems like adding the network name here wouldn't really set us up any better, since the trunk historically has kept the same name when forks occur

**Comment by satish-ravi** [comment]
For the forking use-case, as @cajubelt mentions, at a minimum we'd need to add a new network id for the forked network. So we can always add the new name along with that change. There could be some benefits of having the name remotely configurable, but this actually doesn't seem the right place to do it. The name is a 1:1 mapping to the network id so we don't need to duplicate it for all tokens. I think we can start off with just a static map in the wallet for now, as the change in names for existing network ids (at least what we have currently now) has never happened. We can always make it configurable through a dynamic config mapping network ids to names or even just an entry in the translation file in future releases

**Comment by MuckT** [comment]
@cajubelt @satish-ravi I was hopefully we could support new networks without an app release, but as you mention with networkIds We'd need an app release to update those mappings. With that in mind a static mapping sounds like it will work well.

**Review by MuckT — APPROVED** [review]
I really like that you added schema validation and new tests the computed field `networkIconUrl` 👨‍🍳

File: src/types.ts [hunk 1]
```diff
@@ -41,12 +41,18 @@ export interface TokenInfoJSON {
   pegTo?: string
   isSupercharged?: boolean
   bridge?: string
+  isStableCoin?: boolean
+  isZeroState?: boolean
```
Comment (jeanregisser) [inline src/types.ts:45]: Why do we need this? And would there be a more descriptive name?
```suggestion
  showZeroBalance?: boolean
```

File: README.md [hunk 2]
```diff
@@ -18,13 +18,18 @@ To add a new ERC20 token you need to follow these steps:
...
+| `isStableCoin`      | Whether the token is a stable coin and the price is not expected to change                                                                                                                                                                                   |
+| `isZeroState`       | Whether the token is displayed even if the balance is zero                                                                                                                                                                                                   |
+| `isCashInEligible`  | Whether the token is eligible for cash-ins                                                                                                                                                                                                                   |
+| `isCashOutEligible` | Whether the token is eligible for cash-outs                                                                                                                                                                                                                  |
```
Comment (jeanregisser) [inline README.md:31]: Do we expect outside contributor to fill these? 
The documented fields in the README were only for the required fields to keep external contributions as simple as possible.

**Review by jeanregisser — COMMENTED** [review]
Great!

File: src/types.ts (hunk 1 above)
Comment (satish-ravi) [reply src/types.ts]: This is for showing the token even if the balance is zero in the new [Assets screen](https://www.figma.com/file/OE6RFzqXAMqU9ygHtNYyof/Assets-Page-(Multichain)?node-id=2099%3A17616&mode=dev). And I've updated the name

File: README.md (hunk 2 above)
Comment (satish-ravi) [reply README.md]: Nope, these are internal only. I've removed these and included some of these in types.ts where the name isn't completely obvious.

**Review by jophish — APPROVED** [review]

**Pull request merged**